cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Transaction/RegisterDeregisterStakeAddress.hs

```diff
@@ -0,0 +1,194 @@
+
+  TestnetRuntime
+    { testnetMagic
+    , testnetNodes
+    , wallets=wallet0:wallet1:_
+    , configurationFile
+    }
+    <- cardanoTestnetDefault fastTestnetOptions shelleyOptions conf
+
+  node <- H.headM testnetNodes
+  poolSprocket1 <- H.noteShow $ nodeSprocket node
+  execConfig <- mkExecConfig tempBaseAbsPath poolSprocket1 testnetMagic
+  let socketPath = nodeSocketPath node
+
+  epochStateView <- getEpochStateView configurationFile socketPath
+
+  H.note_ $ "Sprocket: " <> show poolSprocket1
+  H.note_ $ "Abs path: " <> tempAbsBasePath'
+  H.note_ $ "Socketpath: " <> unFile socketPath
+  H.note_ $ "Foldblocks config file: " <> unFile configurationFile
+
+  -- Create Conway constitution
+  gov <- H.createDirectoryIfMissing $ work </> "governance"
+
+  let stakeVkeyFp = gov </> "stake.vkey"
+      stakeSKeyFp = gov </> "stake.skey"
+      stakeCertFp = gov </> "stake.regcert"
+      stakeCertDeregFp = gov </> "stake.deregcert"
+      stakeKeys = KeyPair { verificationKey = File stakeVkeyFp
+                          , signingKey = File stakeSKeyFp
+                          }
+
+  cliStakeAddressKeyGen stakeKeys
+
+  keyDepositStr <- show . unCoin <$> getKeyDeposit epochStateView ceo
+  -- Register stake address
+  void $ execCli' execConfig
+    [ eraName, "stake-address", "registration-certificate"
+    , "--stake-verification-key-file", stakeVkeyFp
+    , "--key-reg-deposit-amt", keyDepositStr
+    , "--out-file", stakeCertFp
+    ]
+
+  stakeAddress <- H.noteM $ execCli' execConfig
+    [ eraName, "stake-address", "build"
+    , "--stake-verification-key-file", stakeVkeyFp
+    , "--out-file", "/dev/stdout"
+    ]
+
+  stakeCertTxBodyFp <- H.note $ work </> "stake.registration.txbody"
+  stakeCertTxSignedFp <- H.note $ work </> "stake.registration.tx"
+
+  txin1 <- findLargestUtxoForPaymentKey epochStateView sbe wallet0
+
+  (_, stdout, stderr) <- execCliAny execConfig
+    [ eraName, "transaction", "build"
+    , "--change-address", Text.unpack $ paymentKeyInfoAddr wallet0
+    , "--tx-in", Text.unpack $ renderTxIn txin1
+    , "--tx-out", Text.unpack (paymentKeyInfoAddr wallet1) <> "+" <> show @Int 10_000_000
+    , "--certificate-file", stakeCertFp
+    , "--witness-override", show @Int 2
+    , "--out-file", stakeCertTxBodyFp
+    ]
+  H.note_ stdout
+  H.note_ stderr
+
+  void $ execCli' execConfig
+    [ eraName, "transaction", "sign"
+    , "--tx-body-file", stakeCertTxBodyFp
+    , "--signing-key-file", signingKeyFp $ paymentKeyInfoPair wallet0
+    , "--signing-key-file", stakeSKeyFp
+    , "--out-file", stakeCertTxSignedFp
+    ]
+
+  void $ execCli' execConfig
+    [ eraName, "transaction", "submit"
+    , "--tx-file", stakeCertTxSignedFp
+    ]
+
+  H.noteShowM_ $ waitForBlocks epochStateView 1
+  -- H.noteShowM_ $ waitForEpochs epochStateView (EpochInterval 2)
+
+  do
+    (_, stdout', stderr') <- execCliAny execConfig
+      [ eraName, "query", "stake-address-info"
+      , "--address", stakeAddress
```

Comment:
We should confirm that this command  succeeds with the expected output.

Reply:
This one was actually a leftover from debugging. It's removed in the final version. The query itself should be tested in `Cardano.Testnet.Test.Cli.Query`, which currently is not (needs fixing).

Reply:
Added check in: https://github.com/IntersectMBO/cardano-node/pull/6026